
<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta charset="utf-8" />
    <title>Genome browser &#8212; Nucleome Browser v0.9.3 documentation</title>
    <link rel="stylesheet" href="_static/nb.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="_static/language_data.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="3D structure viewer" href="component_3d.html" />
    <link rel="prev" title="Panel-container system" href="design.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <div class="section" id="genome-browser">
<span id="components"></span><h1>Genome browser<a class="headerlink" href="#genome-browser" title="Permalink to this headline">¶</a></h1>
<p>Genome browser web component works as conventional genome browsers such as the UCSC Genome Browser, providing visualization support for commonly used genomic data including bigWig, bigBed, tabix, and .hic format.</p>
<div class="section" id="visualization-modes">
<h2>Visualization modes<a class="headerlink" href="#visualization-modes" title="Permalink to this headline">¶</a></h2>
<p><a class="reference internal" href="#gbrowser-toolbar"><span class="std std-numref">Fig. 26</span></a> summarizes the configuration buttons of the genome browser panel in the top toolbar.
Notably, since the Nucleome Browser supports synchronization across multiple panels, two additional navigation modes are introduced in the genome browser web components, i.e., <code class="docutils literal notranslate"><span class="pre">Map</span></code> mode and <code class="docutils literal notranslate"><span class="pre">Context</span></code> mode.
Users can choose a visualization mode by clicking the <code class="docutils literal notranslate"><span class="pre">visualization</span> <span class="pre">mode</span> <span class="pre">button</span></code> (<a class="reference internal" href="_images/icon-genome-mode-normal.png"><img alt="gb-mode" src="_images/icon-genome-mode-normal.png" style="height: 14px;" /></a>) on the right of the genome browser toolbar.</p>
<div class="figure align-center" id="gbrowser-toolbar" style="width: 640px">
<img alt="_images/ch3_gbrowser_toolbar_v2.png" src="_images/ch3_gbrowser_toolbar_v2.png" />
<p class="caption"><span class="caption-number">Fig. 26 </span><span class="caption-text">Genome browser web component toolbar</span><a class="headerlink" href="#gbrowser-toolbar" title="Permalink to this image">¶</a></p>
</div>
<div class="section" id="normal-mode">
<h3>Normal mode<a class="headerlink" href="#normal-mode" title="Permalink to this headline">¶</a></h3>
<p>If a genome browser web component is in the normal mode (<a class="reference internal" href="_images/icon-genome-mode-normal.png"><img alt="gb-mode-normal" src="_images/icon-genome-mode-normal.png" style="height: 14px;" /></a>), it will automatically update itself according to operations dispatched from other web components (for genome browser component they must also be in normal or context mode).
For example, when a user navigates to or highlights a genomic region in other web components, this genome browser component will also go to that region or show the same highlighted region.
Conversely, any operations that happened in this component will broadcast simultaneously to other components.
This mode is quite useful when you want to compare data hosted in different panels side-by-side.</p>
</div>
<div class="section" id="context-mode">
<h3>Context mode<a class="headerlink" href="#context-mode" title="Permalink to this headline">¶</a></h3>
<p><code class="docutils literal notranslate"><span class="pre">Context</span> <span class="pre">mode</span></code> is quite similar to the <code class="docutils literal notranslate"><span class="pre">Normal</span> <span class="pre">mode</span></code>, except that users can set a zooming factor larger than 1x (e.g., 2x, 4x, etc.).
If the zooming factor is 1x, the <code class="docutils literal notranslate"><span class="pre">Context</span> <span class="pre">mode</span></code> will act the same as the <code class="docutils literal notranslate"><span class="pre">Normal</span> <span class="pre">mode</span></code>.
However, when a genome browser component has a zooming factor larger than 1x, this component will automatically zoom out by this scale factor relative to other components.
For example, if other panels currently navigate to a 100kb region (e.g., chr1:10Mb-10.1Mb), a genome component with an 8x zooming factor will navigate to the 800kb region (chr1:9.65Mb-10.45Mb) centered on this 100kb region.
To help users view the relationship between the default region and the zoomed-out region, a light green transparent box will be shown in this genome browser to highlight the region on which other panels are viewing at.
This mode is quite useful when you want to capture the big-picture of a region-of-interest.
For example, you can use one panel to visualize the details of a ChIP-seq peak and use another panel to reveal the context of the peak region without zoom-in and zoom-out back-and-forth.</p>
</div>
<div class="section" id="map-mode">
<h3>Map mode<a class="headerlink" href="#map-mode" title="Permalink to this headline">¶</a></h3>
<p>Finally, you can turn off the synchronization of a genome browser component by clicking the button of navigation mode until the icon becomes <a class="reference internal" href="_images/icon-genome-mode-map.png"><img alt="gb-mode-map" src="_images/icon-genome-mode-map.png" style="height: 14px;" /></a>.
In the <code class="docutils literal notranslate"><span class="pre">Map</span> <span class="pre">mode</span></code>, this panel will not respond to any operations that happened in other panels, and will also not automatically broadcast its operations to other panels.
However, users can still highlight a region in this panel and clicking the <code class="docutils literal notranslate"><span class="pre">go-to</span> <span class="pre">button</span></code> (<a class="reference internal" href="_images/icon-genome-go.png"><img alt="gb-goto" src="_images/icon-genome-go.png" style="height: 14px;" /></a>) to force all other panels to go to the highlighted region.
This mode is useful when you want to use one panel to view the big picture and examine details in other panels.</p>
</div>
</div>
<div class="section" id="navigate-genome">
<h2>Navigate genome<a class="headerlink" href="#navigate-genome" title="Permalink to this headline">¶</a></h2>
<div class="section" id="type-region-s-manually">
<h3>Type region(s) manually<a class="headerlink" href="#type-region-s-manually" title="Permalink to this headline">¶</a></h3>
<p>You can navigate to a certain region(s) by manually typing the genomic coordinate(s) in the genomic coordinate box located on the left of the toolbar.
The genome browser component can recognize any genomic coordinate formatted as chromosome:start-end, such as <code class="docutils literal notranslate"><span class="pre">chr1:1000-2000</span></code>.
Notably, it will assume the coordinates are 1-base and right-fully closed.
It is also possible to view multiple regions together.
Currently, you can type in at most five different genomic regions using a semicolon as the separator (e.g., <code class="docutils literal notranslate"><span class="pre">chr1:1-20000;chr2:1-30000</span></code>).
To view the entire chromosome, you can just type the name of a chromosome (e.g., <code class="docutils literal notranslate"><span class="pre">chr1</span></code> for viewing the whole chromosome 1, <code class="docutils literal notranslate"><span class="pre">chr1;chr2</span></code> for viewing chromosome 1 and chromosome 2 together).
Similarly, at most five different chromosomes can be viewed at the same time.
Nucleome Browser allows users to search for a gene by its name (requiring the gene annotation track).
You can type the gene’s name in the genomic coordinates box and choose the gene in the drop-down list.</p>
</div>
<div class="section" id="navigation-buttons">
<h3>Navigation buttons<a class="headerlink" href="#navigation-buttons" title="Permalink to this headline">¶</a></h3>
<p>You can use four buttons to quickly navigate through the genome.
Clicking the <code class="docutils literal notranslate"><span class="pre">zoom-out</span> <span class="pre">button</span></code> (<a class="reference internal" href="_images/icon-genome-zoomout-3x.png"><img alt="gb-zoom-out" src="_images/icon-genome-zoomout-3x.png" style="height: 14px;" /></a>) will zoom out to a 3x length of the current region.
Clicking the <code class="docutils literal notranslate"><span class="pre">zoom-in</span> <span class="pre">button</span></code> (<a class="reference internal" href="_images/icon-genome-zoomin-3x.png"><img alt="gb-zoom-in" src="_images/icon-genome-zoomin-3x.png" style="height: 14px;" /></a>) will zoom in to a 1/3x length of the current region.
Clicking the <code class="docutils literal notranslate"><span class="pre">move</span> <span class="pre">forward</span> <span class="pre">button</span></code> (<a class="reference internal" href="_images/icon-genome-forward.png"><img alt="gb-forward" src="_images/icon-genome-forward.png" style="height: 14px;" /></a>) will move the currently viewing region to the right for a 1/2x length of the current region.
Clicking the <code class="docutils literal notranslate"><span class="pre">move</span> <span class="pre">backward</span> <span class="pre">button</span></code> (<a class="reference internal" href="_images/icon-genome-backward.png"><img alt="gb-backward" src="_images/icon-genome-backward.png" style="height: 14px;" /></a>) will move the currently viewing region to the left for a 1/2x length of the current region.
Notably, multiple regions overlapping with each other will be automatically merged when you zoom out.</p>
<div class="figure align-center" id="id1" style="width: 480px">
<img alt="_images/ch3_gbrowser_navigate.png" src="_images/ch3_gbrowser_navigate.png" />
<p class="caption"><span class="caption-number">Fig. 27 </span><span class="caption-text">Use the genomic coordinate box to go to a certain region or use the navigation buttons to move along the genome and zoom-in/-out</span><a class="headerlink" href="#id1" title="Permalink to this image">¶</a></p>
</div>
<div class="figure align-center" id="id2" style="width: 420px">
<img alt="_images/ch3_gbrowser_navigate_example.png" src="_images/ch3_gbrowser_navigate_example.png" />
<p class="caption"><span class="caption-number">Fig. 28 </span><span class="caption-text">Illustration of the change of view using different navigation buttons</span><a class="headerlink" href="#id2" title="Permalink to this image">¶</a></p>
</div>
</div>
<div class="section" id="highlight-region-s">
<h3>Highlight region(s)<a class="headerlink" href="#highlight-region-s" title="Permalink to this headline">¶</a></h3>
<p>You can left-click the mouse and drag it on tracks to highlight a particular region.
When you drag it on the Hi-C 2D matrix track, two regions will be highlighted.
After you release the mouse, you can then left-click on the highlight region(s) and drag it to move it along the genome.
Notably, when you highlight a region or move the highlighted region in one panel, other panels will also show the highlighted region as a transparent block simultaneously.
This synchronization also works for highlights on multiple regions.</p>
<div class="figure align-center" id="id3" style="width: 640px">
<img alt="_images/ch3_gbrowser_highlight_v2.png" src="_images/ch3_gbrowser_highlight_v2.png" />
<p class="caption"><span class="caption-number">Fig. 29 </span><span class="caption-text">Right-click the mouse to zoom into any highlighted track or Hi-C contact matrix</span><a class="headerlink" href="#id3" title="Permalink to this image">¶</a></p>
</div>
<p>Right-clicking any place on the highlighted region, you will zoom into the region of highlight.
If a bed track shows categorical data with different colors (e.g., SPIN states, Hi-C subcomponents), you can highlight all the regions with the same annotation/color by left-clicking one region.
Right-clicking on these highlighted regions, a zoom-in view of the highlighted regions will appear.
Notably, multiple regions separated from each other on the genome will be pseudo-merged together as shown below.</p>
<div class="figure align-center" id="id4" style="width: 640px">
<img alt="_images/ch3_gbrowser_multiregion.png" src="_images/ch3_gbrowser_multiregion.png" />
<p class="caption"><span class="caption-number">Fig. 30 </span><span class="caption-text">If multiple regions are highlighted, right-clicking those regions will zoom into regions by stitching discontiguous regions together.</span><a class="headerlink" href="#id4" title="Permalink to this image">¶</a></p>
</div>
</div>
<div class="section" id="use-chromosome-ideogram">
<h3>Use chromosome ideogram<a class="headerlink" href="#use-chromosome-ideogram" title="Permalink to this headline">¶</a></h3>
<p>Chromosome ideogram shows an overview of a chromosome.
The currently viewed region is shown as a red bar just below the chromosome ideogram.
You can left-click the red bar and drag it to quickly navigate to another region in the same chromosome.
You can also brush on the chromosome ideogram to highlight a region and right-click the highlighted region to zoom into that region.</p>
<div class="figure align-center" id="id5" style="width: 480px">
<img alt="_images/ch3_gbrowser_ideogram.png" src="_images/ch3_gbrowser_ideogram.png" />
<p class="caption"><span class="caption-number">Fig. 31 </span><span class="caption-text">Navigate the genome using the chromosome ideogram</span><a class="headerlink" href="#id5" title="Permalink to this image">¶</a></p>
</div>
</div>
</div>
<div class="section" id="export-browser-view">
<h2>Export browser view<a class="headerlink" href="#export-browser-view" title="Permalink to this headline">¶</a></h2>
<div class="section" id="export-screenshot">
<h3>Export screenshot<a class="headerlink" href="#export-screenshot" title="Permalink to this headline">¶</a></h3>
<p>A save-to-png button (<a class="reference internal" href="_images/icon-genome-screenshot.png"><img alt="gb-screenshot" src="_images/icon-genome-screenshot.png" style="height: 14px;" /></a>) can directly export the screenshot of the current panel into a png image file.
You can also create a high-quality image of the current genome browser’s view including the highlights using the print button (<a class="reference internal" href="_images/icon-genome-print.png"><img alt="gb-print" src="_images/icon-genome-print.png" style="height: 14px;" /></a>) in the genome browser toolbar.
You can choose a file format for the downloaded image from pixel-based graphics (png format) and vector-based graphics (SVG, pdf).
Vector-based graphics can be further edited for publication using the edit tool provided on the Nucleome Browser or commercial software such as Adobe Illustrator.
Notably, the print function can only save the current panel into an image file.
You need to save views for different genome browser panels one-by-one.</p>
<div class="figure align-center" id="id6" style="width: 480px">
<img alt="_images/ch3_gbrowser_print.png" src="_images/ch3_gbrowser_print.png" />
<p class="caption"><span class="caption-number">Fig. 32 </span><span class="caption-text">Print the current view to png or SVG file</span><a class="headerlink" href="#id6" title="Permalink to this image">¶</a></p>
</div>
</div>
<div class="section" id="remove-guidelines">
<h3>Remove guidelines<a class="headerlink" href="#remove-guidelines" title="Permalink to this headline">¶</a></h3>
<p>Clicking the <code class="docutils literal notranslate"><span class="pre">remove</span> <span class="pre">guidelines</span> <span class="pre">button</span></code> (<a class="reference internal" href="_images/icon-genome-guideline.png"><img alt="gb-guideline" src="_images/icon-genome-guideline.png" style="height: 14px;" /></a>), you can remove the vertical blue lines.</p>
<div class="figure align-center" id="id7" style="width: 480px">
<img alt="_images/ch3_gbrowser_guide_line.png" src="_images/ch3_gbrowser_guide_line.png" />
<p class="caption"><span class="caption-number">Fig. 33 </span><span class="caption-text">Remove guidelines on the background</span><a class="headerlink" href="#id7" title="Permalink to this image">¶</a></p>
</div>
</div>
</div>
<div class="section" id="configure-tracks">
<h2>Configure tracks<a class="headerlink" href="#configure-tracks" title="Permalink to this headline">¶</a></h2>
<div class="section" id="configure-a-single-track">
<h3>Configure a single track<a class="headerlink" href="#configure-a-single-track" title="Permalink to this headline">¶</a></h3>
<p>Right-clicking on one track label on the left, and choosing the <code class="docutils literal notranslate"><span class="pre">config</span> <span class="pre">button</span></code> from the drop-down list, you will see the configuration dialog.
You can then customize the appearance of a track by modifying the configuration.
Some explanations of settings are shown below:</p>
<ul class="simple">
<li><p><strong>alias</strong>: Set an alternate label for this track, which will be shown on the left of the track.</p></li>
<li><p><strong>color</strong>: Select the primary color for a track.</p></li>
<li><p><strong>height</strong>: Set track’s height (bigWig only).</p></li>
<li><p><strong>mode</strong>: Choose a display mode for a bigWig or bigBed track from <code class="docutils literal notranslate"><span class="pre">full</span></code> and <code class="docutils literal notranslate"><span class="pre">dense</span></code>.</p></li>
<li><p><strong>autoscale</strong>: Whether to automatically scale the min and max value for the bigWig track.</p></li>
<li><p><strong>max</strong>: When <code class="docutils literal notranslate"><span class="pre">autoscale</span></code> is off, set the upper limit of the bigWig track.</p></li>
<li><p><strong>min</strong>: When <code class="docutils literal notranslate"><span class="pre">autoscale</span></code> is off, set the lower limit of the bigWig track.</p></li>
<li><p><strong>norm</strong>: Select normalization method for .hic matrix.</p></li>
<li><p><strong>oe</strong>: Whether to display observed vs expected (O/E) contact matrix rather than observed matrix for .hic data.</p></li>
<li><p><strong>min_bp</strong>: Set the minimum resolution for a Hi-C contact matrix.</p></li>
</ul>
<div class="figure align-center" id="id8" style="width: 640px">
<img alt="_images/ch3_gbrowser_track_config.png" src="_images/ch3_gbrowser_track_config.png" />
<p class="caption"><span class="caption-number">Fig. 34 </span><span class="caption-text">Configure a single track</span><a class="headerlink" href="#id8" title="Permalink to this image">¶</a></p>
</div>
<p>To hide a track, you can right-click on the track label and click the <code class="docutils literal notranslate"><span class="pre">hide</span> <span class="pre">button</span></code> from the drop-down list.
To re-order a track, you can drag that track and move it up or down to the preferred location.</p>
</div>
<div class="section" id="batch-configuration-tool">
<h3>Batch-configuration tool<a class="headerlink" href="#batch-configuration-tool" title="Permalink to this headline">¶</a></h3>
<p>You can also configure a series of tracks together using the batch-configuration tool.
Clicking the <code class="docutils literal notranslate"><span class="pre">batch-configuration</span> <span class="pre">button</span></code> (<a class="reference internal" href="_images/icon-genome-batch.png"><img alt="gb-batch-config" src="_images/icon-genome-batch.png" style="height: 14px;" /></a>) in the genome browser panel toolbar, you will see the dialog of the batch-configuration tool.
In this tool, you can select multiple tracks (holding the <code class="docutils literal notranslate"><span class="pre">Control</span></code> key to add a track one-by-one or the <code class="docutils literal notranslate"><span class="pre">Shift</span></code> key to select a range of tracks) and modify their appearance at the same time.
Notably, only bigWig tracks are supported in this tool.
This tool is quite useful to convert a large number of tracks into dense mode or set the same color for those tracks.</p>
<div class="figure align-center" id="id9" style="width: 640px">
<img alt="_images/ch3_gbrowser_bigwig_batch.png" src="_images/ch3_gbrowser_bigwig_batch.png" />
<p class="caption"><span class="caption-number">Fig. 35 </span><span class="caption-text">Batch configure multiple bigWig tracks</span><a class="headerlink" href="#id9" title="Permalink to this image">¶</a></p>
</div>
<p>We also provide two buttons to directly convert all bigwig tracks from the full view to a compact view and vice versa.
You can access these buttons in the toolbar of the genome browser web component.</p>
<div class="figure align-center" style="width: 640px">
<img alt="_images/ch3_gbrowser_bigwig_full_compact.png" src="_images/ch3_gbrowser_bigwig_full_compact.png" />
</div>
</div>
</div>
<div class="section" id="manage-tracks">
<h2>Manage tracks<a class="headerlink" href="#manage-tracks" title="Permalink to this headline">¶</a></h2>
<p>Clicking the <code class="docutils literal notranslate"><span class="pre">configuration</span> <span class="pre">button</span></code> (<a class="reference internal" href="_images/icon-genome-config.png"><img alt="gb-config" src="_images/icon-genome-config.png" style="height: 14px;" /></a>) on the toolbar of the genome browser panel, you will see the configuration interface.
You can also click the <code class="docutils literal notranslate"><span class="pre">panel-configuration</span> <span class="pre">button</span></code> (<a class="reference internal" href="_images/icon-panel-config_v2.png"><img alt="panel-config" src="_images/icon-panel-config_v2.png" style="height: 14px;" /></a>) on the top-right of the panel to enter the configuration interface.
In the configuration interface, you will see the web page is separated into three parts: 1) data service module on the top; 2) currently loaded tracks shown on the left; 3) available tracks from data service on the right.</p>
<p>In the data service module, you can add data service to the existing list of genomic data services.</p>
<p>In the currently loaded track module, you can quickly re-order tracks by dragging a track, view meta-information for a track, and super-impose a track on a 3D structure (see the 3D structure web component below).</p>
<p>In the available tracks module, you can select a datasheet and add tracks to the list of loaded tracks.
You can filter tracks by clicking the <code class="docutils literal notranslate"><span class="pre">search</span> <span class="pre">button</span></code> and type in the keywords.
Clicking the <code class="docutils literal notranslate"><span class="pre">read-more</span> <span class="pre">button</span></code>, you will be directed to a new website showing extra information about this track (for 4DN data, this leads to the meta-information website on the DCIC data portal).</p>
<div class="figure align-center" id="id10" style="width: 640px">
<img alt="_images/ch3_gbrowser_config.png" src="_images/ch3_gbrowser_config.png" />
<p class="caption"><span class="caption-number">Fig. 36 </span><span class="caption-text">Add or remove tracks in the configuration interface of the genome browser component</span><a class="headerlink" href="#id10" title="Permalink to this image">¶</a></p>
</div>
</div>
<div class="section" id="scatterplot-tool">
<h2>Scatterplot tool<a class="headerlink" href="#scatterplot-tool" title="Permalink to this headline">¶</a></h2>
<p>Nucleome Browser provides a convenient scatterplot tool to interactively compare signals between two bigWig tracks.
Clicking the <code class="docutils literal notranslate"><span class="pre">scatterplot</span> <span class="pre">button</span></code> (<a class="reference internal" href="_images/icon-genome-scatterplot.png"><img alt="gb-scatterplot" src="_images/icon-genome-scatterplot.png" style="height: 14px;" /></a>), a window will appear on the right side of the genome browser panel.
After you select which tracks to be shown in the X-axis and Y-axis, an interactive scatterplot will be shown.
In this scatterplot, each dot indicates a genomic bin (the size of the genomic bin will be automatically adjusted based on the size of the currently viewed region).
Notably, if you highlight regions on genomic tracks or the ideogram of a chromosome, corresponding dots will also be highlighted (by a different color) in the scatterplot.
When you drag the highlighted region, highlighted dots will automatically update.
Conversely, when you use the rectangle or lasso selection tool to manually select a set of dots on the scatterplot, those regions will be highlighted on the tracks as well.</p>
<p>This tool is quite useful to facilitate researchers to discover interesting regions showing an unexpected relationship between two signals.
For other functions of the scatterplot tool such as pan, zoom, reset, etc., you can view the documentation on the Plotly website (<a class="reference external" href="https://plotly.com">https://plotly.com</a>).</p>
<div class="figure align-center" id="id11" style="width: 640px">
<img alt="_images/ch3_gbrowser_scatterplot_v2.png" src="_images/ch3_gbrowser_scatterplot_v2.png" />
<p class="caption"><span class="caption-number">Fig. 37 </span><span class="caption-text">Use the scatterplot tool to explore the relationship between two bigWig tracks</span><a class="headerlink" href="#id11" title="Permalink to this image">¶</a></p>
</div>
</div>
</div>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="index.html">Nucleome Browser</a></h1>








<p class="caption"><span class="caption-text">TUTORIALS</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="animation.html">Animations</a><ul>
<li class="toctree-l2"><a class="reference internal" href="animation.html#overview">Overview</a></li>
<li class="toctree-l2"><a class="reference internal" href="animation.html#panel-organization">Panel organization</a></li>
<li class="toctree-l2"><a class="reference internal" href="animation.html#genome-browser-tutorial">Genome browser tutorial</a></li>
<li class="toctree-l2"><a class="reference internal" href="animation.html#d-structure-tutorial">3D structure tutorial</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="dcic.html">Use 4DN DCIC data</a><ul>
<li class="toctree-l2"><a class="reference internal" href="dcic.html#overview">Overview</a></li>
<li class="toctree-l2"><a class="reference internal" href="dcic.html#use-4dn-public-data">Use 4DN public data</a></li>
<li class="toctree-l2"><a class="reference internal" href="dcic.html#explore-microscopy-data">Explore microscopy data</a></li>
</ul>
</li>
</ul>
<p class="caption"><span class="caption-text">GETTING STARTED</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="design.html">Panel-container system</a><ul>
<li class="toctree-l2"><a class="reference internal" href="design.html#overview">Overview</a></li>
<li class="toctree-l2"><a class="reference internal" href="design.html#what-is-a-panel">What is a panel?</a></li>
<li class="toctree-l2"><a class="reference internal" href="design.html#create-a-new-panel">Create a new panel</a></li>
<li class="toctree-l2"><a class="reference internal" href="design.html#basic-operations">Basic operations</a></li>
<li class="toctree-l2"><a class="reference internal" href="design.html#flexible-layout">Flexible layout</a></li>
</ul>
</li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Genome browser</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#visualization-modes">Visualization modes</a></li>
<li class="toctree-l2"><a class="reference internal" href="#navigate-genome">Navigate genome</a></li>
<li class="toctree-l2"><a class="reference internal" href="#export-browser-view">Export browser view</a></li>
<li class="toctree-l2"><a class="reference internal" href="#configure-tracks">Configure tracks</a></li>
<li class="toctree-l2"><a class="reference internal" href="#manage-tracks">Manage tracks</a></li>
<li class="toctree-l2"><a class="reference internal" href="#scatterplot-tool">Scatterplot tool</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="component_3d.html">3D structure viewer</a><ul>
<li class="toctree-l2"><a class="reference internal" href="component_3d.html#two-visualization-modes">Two visualization modes</a></li>
<li class="toctree-l2"><a class="reference internal" href="component_3d.html#change-atom-style">Change atom style</a></li>
<li class="toctree-l2"><a class="reference internal" href="component_3d.html#exploring-structures">Exploring structures</a></li>
<li class="toctree-l2"><a class="reference internal" href="component_3d.html#configure-color">Configure color</a></li>
<li class="toctree-l2"><a class="reference internal" href="component_3d.html#interaction-across-panels">Interaction across panels</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="component_other.html">Other components</a><ul>
<li class="toctree-l2"><a class="reference internal" href="component_other.html#google-sheet-viewer">Google Sheet viewer</a></li>
<li class="toctree-l2"><a class="reference internal" href="component_other.html#fetch-dna-sequence">Fetch DNA sequence</a></li>
<li class="toctree-l2"><a class="reference internal" href="component_other.html#dn-dcic-imaging-data">4DN DCIC imaging data</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="plugin.html">Plugins</a><ul>
<li class="toctree-l2"><a class="reference internal" href="plugin.html#nucleome-bridge-extension">Nucleome Bridge extension</a></li>
<li class="toctree-l2"><a class="reference internal" href="plugin.html#google-spreadsheet-add-on">Google Spreadsheet add-on</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="session.html">Session Management</a><ul>
<li class="toctree-l2"><a class="reference internal" href="session.html#create-a-session-to-local">Create a session to local</a></li>
<li class="toctree-l2"><a class="reference internal" href="session.html#open-a-saved-session">Open a saved session</a></li>
<li class="toctree-l2"><a class="reference internal" href="session.html#session-management-tool">Session management tool</a></li>
<li class="toctree-l2"><a class="reference internal" href="session.html#share-session">Share session</a></li>
</ul>
</li>
</ul>
<p class="caption"><span class="caption-text">ADVANCED TOPICS</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="data_service.html">Data preparation</a><ul>
<li class="toctree-l2"><a class="reference internal" href="data_service.html#data-type-supported">Data type supported</a></li>
<li class="toctree-l2"><a class="reference internal" href="data_service.html#genomic-data">Genomic data</a></li>
<li class="toctree-l2"><a class="reference internal" href="data_service.html#d-structure-data">3D structure data</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="publish.html">Publish your session</a><ul>
<li class="toctree-l2"><a class="reference internal" href="publish.html#nucleome-browser-gallery">Nucleome Browser Gallery</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="custom.html">Link to customized app</a><ul>
<li class="toctree-l2"><a class="reference internal" href="custom.html#link-to-a-customized-webpage">Link to a customized webpage</a></li>
<li class="toctree-l2"><a class="reference internal" href="custom.html#link-to-dcic-jupyter-notebook">Link to DCIC Jupyter notebook</a></li>
<li class="toctree-l2"><a class="reference internal" href="custom.html#link-to-higlass">Link to HiGlass</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="nb_dispatch_api.html">NB-dispatch API</a><ul>
<li class="toctree-l2"><a class="reference internal" href="nb_dispatch_api.html#overview">Overview</a></li>
<li class="toctree-l2"><a class="reference internal" href="nb_dispatch_api.html#installation">Installation</a></li>
<li class="toctree-l2"><a class="reference internal" href="nb_dispatch_api.html#a-minimal-example">A minimal example</a></li>
<li class="toctree-l2"><a class="reference internal" href="nb_dispatch_api.html#api-reference">API reference</a></li>
</ul>
</li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
      <li>Previous: <a href="design.html" title="previous chapter">Panel-container system</a></li>
      <li>Next: <a href="component_3d.html" title="next chapter">3D structure viewer</a></li>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" />
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2017-2020 Nucleome Browser Authors .
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 2.1.2</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.12</a>
      
      |
      <a href="_sources/component_genome.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>